src/txn/txn.c

```diff
@@ -592,8 +590,29 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 	    __wt_config_gets_def(session, cfg, "commit_timestamp", 0, &cval));
 	if (cval.len != 0) {
 #ifdef HAVE_TIMESTAMPS
-		WT_ERR(__wt_txn_parse_timestamp(
-		    session, "commit", &txn->commit_timestamp, &cval));
+		wt_timestamp_t ts;
+
+		WT_ERR(__wt_txn_parse_timestamp(session, "commit", &ts, &cval));
+		/*
+		 * commit timestamp being set should be newer than the oldest
+		 * timestamp and should move forward in a transaction
+		 */
+		__wt_readlock(session, &txn_global->rwlock);
+		if (txn_global->has_oldest_timestamp && __wt_timestamp_cmp(&ts,
+		    &txn_global->oldest_timestamp) < 0) {
+			__wt_readunlock(session, &txn_global->rwlock);
+			WT_ERR_MSG(session, EINVAL,
+			    "commit timestamp %.*s older than oldest timestamp",
```

Comment:
Do we use `%.*s` to print timestamps anywhere else? I know this is from a configuration string, so it's probably OK, I'm just curious.

Reply:
Yeah, I think we do this everywhere we have config string, I copied from elsewhere.